cmd/kube-apiserver/app/options/features.go

```diff
@@ -0,0 +1,18 @@
+package options
+
+import (
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+var featureGates = map[utilfeature.Feature]utilfeature.FeatureSpec{
+	features.ExternalTrafficLocalOnly: {Default: true, PreRelease: utilfeature.GA},
```

Comment:
If a component registers a previously set feature flag with the same values, the [current behaviour](https://github.com/kubernetes/apiserver/blob/master/pkg/util/feature/feature_gate.go#L168-L170) is to skip it. If there's any deviations, e.g. in terms of conflicting default values, an error is raised.

I think being explicit by making each component register its features would improve the current state of things, where one thing potentially impacts another implicitly.